src/oc/web/images.cljs

```diff
@@ -0,0 +1,42 @@
+(ns oc.web.images
+  (:require [clojure.string :as cstr])
+  (:import [goog Uri]))
+
+(defn- is-filestack?
+  [uri]
+  (= (.getDomain uri) "cdn.filestackcontent.com"))
+
+(defn optimize-filestack-image-url
+  [url preferred-height]
+  (let [uri (Uri. url)]
+    (if-not (is-filestack? uri)
+      url
+      (let [cur-path    (.getPath uri)
+            resize-frag (str "resize=height:" preferred-height)
+            new-path    (str resize-frag "/" cur-path)
+            new-uri     (.setPath uri new-path)]
+        (str new-uri)))))
+
+(defn- is-slack?
+  [uri]
+  (= (.getDomain uri) "avatars.slack-edge.com"))
+
+(defn optimize-slack-image-url
+  [url preferred-height]
+  (let [uri (Uri. url)]
+    (if-not (is-slack? uri)
+      url
+      (let [cur-path (.getPath uri)
+            re       #"_(\d{2,3})\.([a-z]+)$"
+            template (str "_" preferred-height ".$2")
+            new-path (cstr/replace cur-path re template)
+            new-uri  (.setPath uri new-path)]
+        (str new-uri)))))
+
+(defn optimize-image-url
+  [url preferred-height]
+  (let [uri (Uri. url)]
+    (cond
+      (is-filestack? uri) (optimize-filestack-image-url url preferred-height)
+      (is-slack? uri)     (optimize-slack-image-url url preferred-height)
```

Comment:
@cjsauer you check `is-filestack?` and `is-slack?` here and then check again inside the specific function to optimize the url. Seems one check is enough.
You can always apply both optimization if you keep the check inside the specific function (`optimize-slack-image-url` and `optimize-filestack-image-url`).
WYT?

Reply:
@bago2k4 yeah I noticed this too while implementing it. Given that it's essentially just a string comparison, I figured it would be nice for these function to remain useful individually, i.e. it would be nice to be able to call `(optimize-slack-image-url u)` when `u` is _known_ to be a Slack URL.

Reply:
y it makes sense, i was suggesting something like `(-> url (optimize-filestack-image-url preferred-height) (optimize-slack-image-url preferred-height))`.
The check will happen one time inside the function and it will return the initial string if it's not a slack/filestack url.
But no problem, it's only a comparison.